libraries/azurerm_subnet.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzurermSubnet < AzurermSingularResource
+  name 'azurerm_subnet'
+  desc 'Verifies settings for an Azure Virtual Network Subnet'
+  example <<-EXAMPLE
+    describe azurerm_subnet(resource_group: 'example',vnet: 'virtual-network-name' name: 'subnet-name') do
+      it { should exist }
+      its('name') { should eq 'subnet-name' }
+    end
+  EXAMPLE
+
+  ATTRS = %i(
+    id
+    name
+    type
+    properties
+  ).freeze
+
+  attr_reader(*ATTRS)
+
+  def initialize(resource_group: nil, vnet: nil, name: nil)
+    resp = client.subnet(resource_group, vnet, name)
+    return if resp.nil? || resp.key?('error')
```

Comment:
We have a new helper method you may use here: 
```
 return if has_error?(resp)
```
That will check if the response has that key or is nil. Source here: https://github.com/inspec/inspec-azure/blob/master/libraries/azurerm_resource.rb#L46-L48